File: src/lib/scala/Slate.Common/src/main/scala/slate/common/EnvSupport.scala
/**
  * <slate_header>
  * author: Kishore Reddy
  * url: https://github.com/kishorereddy/scala-slate
  * copyright: 2016 Kishore Reddy
  * license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  * desc: a scala micro-framework
  * usage: Please refer to license on github for more info.
  * </slate_header>
  */
package slate.common

trait EnvSupport {


  def isProd:Boolean =
  {
    isEnv(Env.PROD)
  }


  def isUat:Boolean =
  {
    isEnv(Env.UAT)
  }


  def isQa:Boolean =
  {
    isEnv(Env.QA)
  }


  def isDev:Boolean =
  {
    isEnv(Env.DEV)
  }


  def isEnv(envMode: String):Boolean = ???
}

File: src/common/common/src/main/kotlin/kiit/common/envs/EnvSupport.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package kiit.common.envs

interface EnvSupport {

    val isPro: Boolean get() = isEnv(EnvMode.Pro)

    val isUat: Boolean get() = isEnv(EnvMode.Uat)

    val isQat: Boolean get() = isEnv(EnvMode.Qat)

    val isDev: Boolean get() = isEnv(EnvMode.Dev)

    val isDis: Boolean get() = isEnv(EnvMode.Dis)

    val isOther: Boolean get() {
        return !isDev && !isQat && !isUat && !isPro && !isDis
    }

    fun isEnv(envMode: EnvMode): Boolean

    fun isEnvName(envMode: String): Boolean
}
